credentials/src/test/kotlin/SSICredStatusListTest.kt

```diff
@@ -0,0 +1,277 @@
+
+    val credentialStatus: StatusList2021Entry =
+      StatusList2021Entry.fromJsonObject(credWithCredStatus.credentialStatus.jsonObject)
+
+    assertEquals(credentialStatus.id.toString(), "cred-with-status-id")
+    assertEquals(credentialStatus.type.toString(), "StatusList2021Entry")
+    assertEquals(credentialStatus.statusPurpose.toString(), "revocation")
+    assertEquals(credentialStatus.statusListIndex, "123")
+    assertEquals(credentialStatus.statusListCredential.toString(), "status-list-cred-id")
+  }
+
+  @Test
+  fun `happy path generation`() {
+    val revocationID = "revocation-id"
+    val testIssuer = did
+
+    val testCred1 = createTestCredWithStatus(
+      did,
+      "test-subject-id-1",
+      mapOf("firstName" to "Bobby"),
+      "123"
+    )
+
+    val testCred2 = createTestCredWithStatus(
+      did,
+      "test-subject-id-2",
+      mapOf("firstName" to "Joe"),
+      "124"
+    )
+
+    val statusListCredentialJwt =
+      VerifiableCredential.generateStatusList2021Credential(
+        signOptions,
+        revocationID,
+        testIssuer,
+        "revocation",
+        listOf(testCred1, testCred2)
+      )
+
+    val statusListCredential = VerifiableCredential.fromJwt(statusListCredentialJwt)
+
+    assertNotNull(statusListCredential)
+    assertTrue(
+      statusListCredential.contexts.containsAll(
+        listOf(
+          URI.create("https://www.w3.org/2018/credentials/v1"),
+          URI.create("https://w3id.org/vc/status-list/2021/v1")
+        )
+      )
+    )
+    assertTrue(
+      statusListCredential.types.containsAll(
+        listOf(
+          "VerifiableCredential",
+          "StatusList2021Credential"
+        )
+      )
+    )
+    assertEquals(statusListCredential.credentialSubject.id, URI.create(revocationID))
+    assertEquals(statusListCredential.credentialSubject.type, "StatusList2021")
+    assertEquals(
+      "revocation",
+      statusListCredential.credentialSubject.jsonObject["statusPurpose"] as? String?
+    )
+    assertEquals(
+      "H4sIAAAAAAAA/2NgQAESAAPT1/8QAAAA",
+      statusListCredential.credentialSubject.jsonObject["encodedList"] as? String?
+    )
+  }
+
+  @Test
+  fun `fails when duplicate statusListIndex`() {
+    val revocationID = "revocation-id"
+    val testIssuer = did
+
+    val testCred1 = createTestCredWithStatus(
+      did,
+      "test-subject-id-1",
+      mapOf("company" to "Block", "website" to "https://block.xyz"),
+      "123"
+    )
+
+    val testCred2 = createTestCredWithStatus(
+      did,
+      "test-subject-id-1",
+      mapOf("company" to "Block", "website" to "https://block.xyz"),
+      "123"
+    )
+
+    val exception = assertThrows<Exception> {
+      VerifiableCredential.generateStatusList2021Credential(
+        signOptions,
+        revocationID,
+        testIssuer,
+        "revocation",
+        listOf(testCred1, testCred2)
+      )
+    }
+
+    assertTrue(
+      exception
+        .message!!.contains("duplicate entry found")
```

Comment:
Would be useful to have the message contain which entry was the duplicate entry found, and which credentials it belonged to.